src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -2084,27 +2082,36 @@ public function registerNodeTypes($types, $allowUpdate)
                 $query = "SELECT * FROM phpcr_type_nodes WHERE name = ?";
                 $result = $this->getConnection()->fetchColumn($query, array($type->getName()));
                 if ($result) {
-                    $this->getConnection()->delete('phpcr_type_nodes', array('node_type_id' => $result));
                     $this->getConnection()->delete('phpcr_type_props', array('node_type_id' => $result));
                     $this->getConnection()->delete('phpcr_type_childs', array('node_type_id' => $result));
                 }
             }
 
+            $data = array(
+                'name' => $type->getName(),
+                'supertypes' => implode(' ', $type->getDeclaredSuperTypeNames()),
+                'is_abstract' => $type->isAbstract() ? 1 : 0,
+                'is_mixin' => $type->isMixin() ? 1 : 0,
+                'queryable' => $type->isQueryable() ? 1 : 0,
+                'orderable_child_nodes' => $type->hasOrderableChildNodes() ? 1 : 0,
+                'primary_item' => $type->getPrimaryItemName(),
+            );
+
             try {
-                $this->getConnection()->insert('phpcr_type_nodes', array(
-                    'name' => $type->getName(),
-                    'supertypes' => implode(' ', $type->getDeclaredSuperTypeNames()),
-                    'is_abstract' => $type->isAbstract() ? 1 : 0,
-                    'is_mixin' => $type->isMixin() ? 1 : 0,
-                    'queryable' => $type->isQueryable() ? 1 : 0,
-                    'orderable_child_nodes' => $type->hasOrderableChildNodes() ? 1 : 0,
-                    'primary_item' => $type->getPrimaryItemName(),
-                ));
+                if ($allowUpdate && $result) {
+                    $this->getConnection()->update('phpcr_type_nodes', $data, array('node_type_id' => $result));
+                } else {
+                    $this->getConnection()->insert('phpcr_type_nodes', $data);
+                }
```

Comment:
while we clean this up, why not do

```php
if ($result) {
    if (!$allowUpdate) {
        throw ...
    }
    $this->getConnection()->update...
} else {
   $this->getConnection()->insert...
}
```

and get rid of the try-catch. as we know if the result exists or not, we can handle it this way, and not hide internal errors with missing validation and invalid queries as a node type exists..